lib/active_merchant/versionable.rb

```diff
@@ -0,0 +1,29 @@
+module ActiveMerchant
+  module Versionable
+    def self.included(base)
+      if base.respond_to?(:class_attribute)
+        base.class_attribute :versions, default: {}
+        base.extend(ClassMethods)
+      end
+    end
+
+    module ClassMethods
+      def inherited(subclass)
+        super
+        subclass.versions = {}
+      end
+
+      def version(version, feature = :default_api)
```

Comment:
How is this method used? Is it just for the `versionable_test.rb`?

Reply:
This method is inside the ClassMethods module. This module is extended to the base class in line 6, so any methods inside this module will be class methods.

The intended usage for this method is to be used at the top of any Gateway subclass.

```ruby
class MyTestGateway < Gateway
  version 'v3' # This sets the :default_api key in the versions hash.
  version 'v5', :recurring_api # This sets the :recurring_api key in the versions hash
end
```

You can see this method being used in this PR in `adyen.rb`, `braintree_blue.rb` and `global_collect.rb`.